### ISSUE TITLE
Julia hangs on exceptions in HTTP WebSocket block

### ISSUE THREAD

**Issue by thedimlebowski** [JuliaWeb/HTTP.jl#698]
I am new to Julia and am encountering some weird behavior when trying to throw exceptions while listening to a WebSocket stream. For example, the following code snippet will hang forever after printing the connection message:

```
using HTTP

HTTP.WebSockets.open("wss://www.bitmex.com/realtime") do ws
    message = String(readavailable(ws))
    println(message)
    error("This is an error")
end
```

This is also freezing if an undefined variable is being referenced (this is how I first encountered the problem).

After some investigation, it looks like it comes from the HTTP package. The close function hangs because the server keeps sending messages https://github.com/JuliaWeb/HTTP.jl/blob/2a03ca76376162ffc3423ba7f15bd6d966edff9b/src/WebSockets.jl#L237

---

Originally posted @ https://stackoverflow.com/questions/66982506/julia-hangs-on-exceptions-in-http-websocket-block/66985257#66985257

**Comment by thedimlebowski** [comment]
Related to #557

**Comment by quinnj** [comment]
Thanks for the report @thedimlebowski and sorry for the slow response. We lacked any real maintainers of the websockets code until now. Your diagnosis was correct; if any kind of error was thrown from the `WebSockets.open` handler function, it triggered a finally block to be run [here](https://github.com/JuliaWeb/HTTP.jl/blob/8c21c104b7b1e34c403655e180f6ae9dd1122ab3/src/WebSockets.jl#L182) where it attempted to initiate the CLOSE sequence and indeed, if the server wasn't responding, then we would just block forever.

This has been corrected on HTTP#master where I recently merged an overhaul of the websockets code and we now include a timeout when waiting for a server to respond to a CLOSE sequence.
